### PR TITLE
TASK-1272: Removed the ObjectMappers and replaced them with the RestH…

### PR DESCRIPTION
**Pull request by krozzus**
…elper

### CODE REVIEW

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java [hunk 1]
```diff
@@ -50,15 +46,21 @@
 
   private static final ParameterizedTypeReference<TaskanaPagedModel<TaskSummaryRepresentationModel>>
       TASK_SUMMARY_PAGE_MODEL_TYPE =
-          new ParameterizedTypeReference<TaskanaPagedModel<TaskSummaryRepresentationModel>>() {};
+      new ParameterizedTypeReference<TaskanaPagedModel<TaskSummaryRepresentationModel>>() {
+      };
   private static RestTemplate template;
-
+  private final TaskServiceImpl taskService;
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:52]: Pls remove this

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java [hunk 2]
```diff
@@ -50,15 +46,21 @@
...
+  @Autowired
+  RestHelper restHelper;
+  @Autowired
+  private DataSource dataSource;
 
-  @Autowired RestHelper restHelper;
-
-  @Autowired private DataSource dataSource;
+  @Autowired
+  TaskControllerIntTest(TaskServiceImpl taskService) {
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:61]: pls remove this

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java [hunk 3]
```diff
@@ -565,83 +567,44 @@ void testGetQueryByPorSecondPageSortedByType() {
...
+        template.exchange(
+            restHelper.toUrl(Mapping.URL_TASKS_ID,
+                "TKI:000000000000000000000000000000000002"),
+            HttpMethod.GET,
+            new HttpEntity<>(restHelper.getHeadersAdmin()),
+            ParameterizedTypeReference.forType(TaskRepresentationModel.class));
+
+    TaskRepresentationModel repModel = response.getBody();
+    assertThat(repModel).isNotNull();
+    assertThat(repModel.getCreated().matches("\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\.\\d{3}Z"))
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:581]: I think you could use assertThat(xyz).matches(pattern);

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java [hunk 4]
```diff
@@ -50,15 +46,21 @@
 
   private static final ParameterizedTypeReference<TaskanaPagedModel<TaskSummaryRepresentationModel>>
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:47]: we could add one for TaskRepresentationModel as well, to be consistent...

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java [hunk 5]
```diff
@@ -50,15 +46,21 @@
...
+      };
   private static RestTemplate template;
-
+  private final TaskServiceImpl taskService;
   @Value("${taskana.schemaName:TASKANA}")
   public String schemaName;
+  @Autowired
+  RestHelper restHelper;
+  @Autowired
+  private DataSource dataSource;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:58]: Please use an autowired constructor ;)

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java (hunk 3 above)
Comment (mustaphazorgati) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:581]: yes, you can. See [here](http://joel-costigliola.github.io/assertj/core-8/api/org/assertj/core/api/AbstractCharSequenceAssert.html#matches-java.lang.CharSequence-)

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java [hunk 6]
```diff
@@ -565,83 +567,44 @@ void testGetQueryByPorSecondPageSortedByType() {
...
+                "TKI:000000000000000000000000000000000002"),
+            HttpMethod.GET,
+            new HttpEntity<>(restHelper.getHeadersAdmin()),
+            ParameterizedTypeReference.forType(TaskRepresentationModel.class));
+
+    TaskRepresentationModel repModel = response.getBody();
+    assertThat(repModel).isNotNull();
+    assertThat(repModel.getCreated().matches("\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\.\\d{3}Z"))
+        .isTrue();
+    assertThat(repModel.getAttachments().isEmpty()).isFalse();
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:583]: here aswell. `assertThat(repModel.getAttachments()).isEmpty()`

Comment (mustaphazorgati) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:583]: http://joel-costigliola.github.io/assertj/core-8/api/org/assertj/core/api/AbstractIterableAssert.html#isEmpty--

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java [hunk 7]
```diff
@@ -540,96 +533,56 @@ void testGetQueryByPorSecondPageSortedByType() {
...
+    ResponseEntity<TaskRepresentationModel> responseUpdate =
+        template.exchange(
+            restHelper.toUrl(Mapping.URL_TASKS_ID, "TKI:100000000000000000000000000000000000"),
+            HttpMethod.PUT,
+            new HttpEntity<>(originalTask, restHelper.getHeaders()),
+            TASK_MODEL_TYPE);
 
-    assertThat(updatedTaskObject.getModified()).isNotEqualTo(originalTaskObject.getModified());
+    TaskRepresentationModel updatedTask = responseUpdate.getBody();
+    assertThat(originalTask.getModified()).isNotEqualTo(updatedTask.getModified());
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:585]: Better: Check for `isAfter`

**Pull request merged**